lib/howl/interactions/file_selection.moon

```diff
@@ -0,0 +1,302 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+glib = require 'ljglibs.glib'
+
+import app, clipboard, config, interact, log, Project from howl
+import File from howl.io
+import markup, style, ListWidget from howl.ui
+import Matcher from howl.util
+
+append = table.insert
+separator = File.separator
+
+style.define_default 'directory', 'key'
+style.define_default 'filename', 'string'
+
+howl.config.define
+  name: 'hidden_file_extensions'
+  description: 'File extensions that determine which files should be hidden in file selection lists'
+  scope: 'global'
+  type_of: 'string_list'
+  default: {'a', 'bc', 'git', 'hg', 'o', 'pyc', 'so'}
+
+home_dir = -> File glib.get_home_dir!
+
+root_dir = (file) ->
+  if not file
+    return root_dir home_dir!
+  while file.parent
+    file = file.parent
+  return file
+
+display_name = (file, is_directory, base_directory) ->
+  return ".#{separator}" if file == base_directory
+  rel_path = file\relative_to_parent(base_directory)
+  return is_directory and rel_path .. separator or rel_path
+
+parse_path = (path) ->
+  if not path or path.is_blank or not File.is_absolute path
+    local directory
+    if app.editor and app.editor.buffer
+      file = app.editor.buffer.file
+      directory = file.parent if file
+    if not directory
+      directory = home_dir!
+
+    if not path or path.is_blank
+      return directory, ''
+    else
+      path = tostring directory / path
+
+  path = File.expand_path path
+  root_marker = separator .. separator
+  pos = path\rfind root_marker
+  if pos
+    path = path\sub pos + 1
+
+  unmatched = ''
+  local closest_dir
+  while not path.is_empty
+    pos = path\rfind(separator)
+    if pos
+      unmatched = path\sub(pos + 1) .. unmatched
+      path = path\sub 1, pos
+    closest_dir = File path
+    if closest_dir.exists and closest_dir.is_directory
```

Comment:
The `if .. else` can be removed. A `break if closest_dir.is_directory` line with the other code following suffices. The `if not pos\nbreak` below could be also be more readable in my opinion as a one-liner; `break unless pos`.